### PR TITLE
RaceFacadeImpl RaceParticipationFacadeImpl TestDriveFacadeImpl & test's

### PR DESCRIPTION
**Pull request by Adel83**

### CODE REVIEW

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    }
+
+    @Test
+    void findTestDriveByIdTest() {
+        when(beanMappingServiceMock.mapTo(validTestDrive, TestDriveDTO.class)).thenReturn(testDriveDTO);
+        when(testDriveService.findById(3L)).thenReturn(validTestDrive);
+
+        TestDriveDTO resTestDriveDTO = testDriveFacadeImpl.findTestDriveByID(validTestDrive.getId());
+        Assert.assertTrue((resTestDriveDTO).equals(testDriveDTO));
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: User `assertEquals` instead.

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    @Test
+    public void deleteTestDriveTest() {
+
+        Long id = 3L;
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+        when(testDriveService.findById(id)).thenReturn(validTestDrive);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        testDriveFacadeImpl.deleteTestDrive(testDriveDTO);
+
+        verify(testDriveService, times(1)).remove(validTestDrive);
+    }
+
+    @Test
+    public void updateTestDriveTest() {
+
+        Long id = 3L;
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        when(testDriveService.findById(id)).thenReturn(validTestDrive);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+        testDriveFacadeImpl.updateTestDrive(testDriveDTO);
+
+        verify(testDriveService, times(1)).update(validTestDrive);
+    }
+
+    @Test
+    public void addTestDriveTest() {
+
+        Long id = 3L;
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+        when(testDriveService.findById(id)).thenReturn(validTestDrive);
```
Comment (elderanakain) [inline formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/TestDriveFacadeImplTest.java]: Why do you need this?

File: formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/RaceFacadeImplTest.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+        validRaceDTO = new RaceDTO();
+        validRaceDTO.setDate(new Date());
+        validRaceDTO.setLocation("location");
+        validRaceDTO.setTitle("GP Monaco");
+        validRaceDTO.setId(3L);
+
+    }
+
+    protected Race createRace() {
```
[the remark this answers is not in the document]
Comment (Adel83) [reply formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/RaceFacadeImplTest.java]: createRace is from BaseTest.

[the remark this answers is not in the document]
Comment (elderanakain) [reply formulaoneteam-service/src/test/java/cz/muni/fi/pa165/service/facade/RaceFacadeImplTest.java]: You should use this method - not copy paste from there.

File: formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/RaceParticipationFacadeImpl.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        raceParticipationService.remove(beanMappingService.mapTo(raceParticipationDTO, RaceParticipation.class));
+
+    }
+
+    @Override
+    public void addRaceParticipation(RaceParticipationDTO raceParticipationDTO) {
+        if (raceParticipationDTO == null) throw new IllegalArgumentException("null raceParticipationAddDTO, cannot add");
+        RaceParticipation raceParticipation = beanMappingService.mapTo(raceParticipationDTO, RaceParticipation.class);
+        raceParticipationService.add(raceParticipation);
+        raceParticipationDTO.setId(raceParticipation.getId());
```
[the remark this answers is not in the document]
Comment (elderanakain) [reply formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/RaceParticipationFacadeImpl.java]: Still not resolved.

File: formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/RaceFacadeImpl.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        raceService.remove(beanMappingService.mapTo(raceDTO, Race.class));
+    }
+
+    @Override
+    public void addRace(RaceDTO raceDTO) {
+        if (raceDTO == null) throw new IllegalArgumentException("null raceAddDTO, cannot add");
+
+        Race race = beanMappingService.mapTo(raceDTO, Race.class);
+        raceService.add(race);
+        raceDTO.setId(race.getId());
```
[the remark this answers is not in the document]
Comment (elderanakain) [reply formulaoneteam-service/src/main/java/cz/muni/fi/pa165/service/facade/RaceFacadeImpl.java]: Still not resolved.

**Pull request merged**